Userland/Libraries/LibPDF/ColorSpace.cpp

```diff
@@ -101,7 +101,7 @@ NonnullRefPtr<DeviceGrayColorSpace> DeviceGrayColorSpace::the()
     return instance;
 }
 
-PDFErrorOr<Color> DeviceGrayColorSpace::color(ReadonlySpan<Value> arguments) const
+PDFErrorOr<ColorOrStyle> DeviceGrayColorSpace::style(ReadonlySpan<Value> arguments) const
```

Comment:
Another idea could be to make this return a reference to a Style and make the non-Pattern colorspaces could always return the same object and just change the value it returns?

Or we could just special-case Pattern. The callsites are already a mess anyways.

Or we could get back to your original way and worry about perf later.

But as-is is fine too, whatever you think is best.

Reply:
I wasn't too happy with using `SolidColorPaintStyle` everywhere either, it seemed a bit overkill. I think `Variant<Color, PaintStyle>` is probably a natural way to go, doesn't immediately mean perf will be terrible and also doesn't blow up the inheritance hierarchy (which is possibly a bad thing - but at least it mirrors the spec). I think just roll with it for now though, we can always come back and refactor.